db_stress_tool/db_stress_tool.cc

```diff
@@ -62,18 +64,36 @@ int db_stress_tool(int argc, char** argv) {
 
   Env* raw_env;
 
+  int env_opts =
+      !FLAGS_hdfs.empty() + !FLAGS_env_uri.empty() + !FLAGS_fs_uri.empty();
+  if (env_opts > 1) {
+    fprintf(stderr,
+            "Error: --hdfs, --env_uri and --fs_uri are mutually exclusive\n");
+    exit(1);
+  }
+
   if (!FLAGS_hdfs.empty()) {
-    if (!FLAGS_env_uri.empty()) {
-      fprintf(stderr, "Cannot specify both --hdfs and --env_uri.\n");
-      exit(1);
-    }
     raw_env = new ROCKSDB_NAMESPACE::HdfsEnv(FLAGS_hdfs);
   } else if (!FLAGS_env_uri.empty()) {
     Status s = Env::LoadEnv(FLAGS_env_uri, &raw_env, &env_guard);
     if (raw_env == nullptr) {
       fprintf(stderr, "No Env registered for URI: %s\n", FLAGS_env_uri.c_str());
       exit(1);
     }
+  } else if (!FLAGS_fs_uri.empty()) {
+#ifndef OS_WIN
```

Comment:
Why is this in an OS_WIN block?  Can't we load FileSystems on Windows?

Reply:
See the comment in the source below: - // NewCompositeEnv is not available in Windows
Linking would fail on windows if we would preprocess the call out.

Reply:
Looking over this again, I found that there is  is an implementation of NewCompositeEnv for OS_WIN in env/env.cc. I'll add support for the new variant there as well, and db_bench/db_stress should compile just fine for windows.

Reply:
Implemented NewCompositeEnv for windows and removed the ifdef.